packages/commerce-server/src/record-new-purchase.ts

```diff
@@ -28,8 +28,14 @@ export class PurchaseError extends Error {
   }
 }
 
-export async function stripeData(checkoutSessionId: string) {
-  const {getCheckoutSession} = getStripeSdk()
+type StripeDataOptions = {
+  checkoutSessionId: string
+  stripeCtx?: StripeContext
+}
+
+export async function stripeData(options: StripeDataOptions) {
+  const {stripeCtx, checkoutSessionId} = options
+  const {getCheckoutSession} = getStripeSdk({ctx: stripeCtx})
```

Comment:
Updates the `stripeData` function to receive `options` instead of just the `checkoutSessionId` so that an optional `stripeContext` can be passed in. We can pass that down to `getStripeSdk` regardless of whether it is defined, because if it is `undefined`, it will be overridden with the default context.